### PR TITLE
Create UserContext and realm service, useMongo hook

### PR DESCRIPTION
**Pull request by jt-mchorse**
- Auth state now can be delivered through UserContext
```
type UserContextValue = {
  loading: boolean;
  user: Realm.User;
  mongoUserKey: string;
  isAdmin: boolean;
  isLoggedIn: boolean;
  actions: {
    setUserAPIKey: (key: string) => void,
    logout: () => void
  }
}
```

- `runQuery` and `updateOne` API can be accessed through `useMongo` hook.

### CODE REVIEW

**Comment by jt-mchorse** [comment]
@smcgregor , this PR is stacked over [102](https://github.com/PartnershipOnAI/aiid/pull/102), you can review changes [here](https://github.com/jt-mchorse/aiid/pull/2).

**Comment by smcgregor** [comment]
@jt-mchorse I merged in the linting change set. Can you merge or rebase master into this PR to clean up the diff?

**Comment by jt-mchorse** [comment]
Yup

**Comment by jt-mchorse** [comment]
@smcgregor , please review.

File: site/gatsby-site/src/hooks/useMongo.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { realmApp } from '../mongodb/authenticate';
```
Comment (smcgregor) [inline site/gatsby-site/src/hooks/useMongo.js:1]: This looks like it is not the correct path. Isn't this at `site/gatsby-site/src/services/realmApp.js`?

**Comment by smcgregor** [comment]
We are likely to generalize and expand this substantially at some point, but it doesn't need to be now. We just need to make sure it is maintainable and will work more flexibly and completely through time and iteration. Does one of your subsequent MRs use this code so I can see it in action? I would like to see what this looks like from consuming parts of the application.

File: site/gatsby-site/src/hooks/useMongo.js (hunk 1 above)
Comment (jt-mchorse) [reply site/gatsby-site/src/hooks/useMongo.js:1]: oh, how silly auto import, yes, correct.

**Comment by jt-mchorse** [comment]
@smcgregor , of course it does, userContext is used in #77 , #100 . especially, #100  will make all user auth related methods using this context and will remove existing `authenticate.js`

**Comment by smcgregor** [comment]
I am still figuring out how this all integrates and how all these hooks work, but I can see that this does not affect any of the functionality already present and your later feature-centered merge requests depend on this. I may come back with more comments and/or questions, but I will merge this now and focus on the next PR in the chain.

**Pull request merged**